pkg/controller/servicebindingrequest/nested/nested.go

```diff
@@ -0,0 +1,181 @@
+	if !exists {
+		return nil, false, nil
+	}
+
+	return getValue(val, path.Tail())
+}
+
+// collectValues accumulates the values found in all the given `path` in all
+// elements present in `obj`.
+func collectValues(obj []interface{}, path Path) (interface{}, error) {
+	r := accumulator.NewAccumulator()
+	for _, e := range obj {
+		val, found, err := getValue(e, path.AdjustedPath())
+		if err != nil || !found {
+			return nil, err
+		}
+		err = r.Accumulate(val)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return r.Value(), nil
+}
+
+// UnsupportedTypeErr is returned when an unsupported type is encountered.
+var UnsupportedTypeErr = errors.New("unsupported type")
+
+// convertToSlice attempts to convert the given `src` into a `[]interface{}`. An
+// error is returned when `src` is not one of the following:
+//
+// - []map[string]interface{}
+// - []string
+// - []int
+//
+func convertToSlice(src interface{}) ([]interface{}, error) {
+	var obj []interface{}
+	switch t := src.(type) {
+	case []map[string]interface{}:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	case []string:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	case []int:
+		obj = make([]interface{}, len(t))
+		for i, e := range t {
+			obj[i] = e
+		}
+	default:
+		return nil, UnsupportedTypeErr
+	}
+	return obj, nil
+}
+
+// InvalidIndexErr is returned when a given index is out of bounds.
+var InvalidIndexErr = errors.New("invalid index")
+
+// getValueFromSlice attempts to return the value present at the given path.
+func getValueFromSlice(s interface{}, path Path) (interface{}, bool, error) {
+	// assert and convert s to []interface{}
+	obj, err := convertToSlice(s)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// it is required for path to have a head in this case, since it is expected to contain either an
+	// Index or a sub-key in order to extract or aggregate the underlying value.
+	head, ok := path.Head()
+	if !ok {
+		return nil, false, nil
+	}
+
+	if head.Index != nil {
+		if *head.Index > len(obj) {
+			return nil, false, InvalidIndexErr
+		}
+		m := obj[*head.Index]
+		return getValue(m, path.Tail())
+	}
+
+	r, err := collectValues(obj, path)
+	if err != nil {
+		return nil, false, err
+	}
+	return r, true, nil
+}
+
+// getValue attempts to return the value present at the given path.
+func getValue(obj interface{}, path Path) (interface{}, bool, error) {
+	// return obj if path is empty.
+	if _, ok := path.Head(); !ok {
+		return obj, true, nil
+	}
+
+	switch val := obj.(type) {
+	case string, int: // scalar
+		if path.HasTail() {
+			return nil, false, fmt.Errorf("type doesn't accept an index or key")
+		}
+		return val, true, nil
+	case map[string]interface{}: // map
+		return getValueFromMap(val, path)
+	case []map[string]interface{}, []int, []string: // slice
+		return getValueFromSlice(val, path)
+	default:
+		panic(fmt.Sprintf("missing type for %+v", val))
```

Comment:
Instead of `panic` proper error should be returned.
What about `UnsupportedTypeErr`?

Reply:
I left this `panic` there to indicate the supported types cases is incomplete. Perhaps some other types are missing such as  `int64` and `float64` since the JSON decoder produces those.